src/weather.cpp

```diff
@@ -146,7 +146,12 @@ weather_sum sum_conditions( const time_point &start, const time_point &end,
             tick_size = 1_minutes;
         }
 
-        const auto wtype = wgen.get_weather_conditions( location, to_turn<int>( t ), g->get_seed() );
+        weather_type wtype;
+        if( g->weather_override == WEATHER_NULL ) {
+            wtype = wgen.get_weather_conditions( location, to_turn<int>( t ), g->get_seed() );
```

Comment:
I believed you don't need that int cast - `t` is of type `timepoint` which is expected by `weather_generator::get_weather_conditions`

```suggestion
            wtype = wgen.get_weather_conditions( location, t, g->get_seed() );
```